docs-src/references/configuration.md

```diff
@@ -387,3 +387,123 @@ publicClient:
 ```
 
 Use `dns:///` prefix to enable round-robin between IP address for DNS name.
+
+## archival
+
+_Optional_
+
+Archival is an optional configuration needed to set up the [Archival store](/concepts/what-is-archival).
+It can be enabled on the `history` and `visibility`.
+
+The following list describes supported values for each configuration on the `history` and `visibility` data.
+
+- `state` : Supported values are `enabled`, `disabled` or `paused`. This value must be `enabled` to use Archival with any Namespace in your Cluster.
+- `enableRead`: Supported values are `true` or `false`. Set to `true` to allow read operations from the archived Event History data.
+- `provider`: Location where data should be archived. Subprovider configs are `filestore`, `gstorage`, `s3`, or `your_custom_provider`. Default configuration specifies `filestore`.
+
```

Comment:
There's very strict validating rules for archival & namespaceDefault config: https://github.com/temporalio/temporal/blob/master/common/config/archival.go#L53

If `state` is enabled: provider section and URI in namespaceDefaults must be provided.
If `state` is disabled: enableRead must be false; state in namespaceDefault must be disabled; provider and URI in namespaceDefault must not be set

And example for disabling archival is probably useful here:
```
archival:
  history:
    state: "disabled"
    enableRead: false
  visibility:
    state: "disabled"
    enableRead: false

namespaceDefaults:
  archival:
    history:
      state: "disabled"
    visibility:
      state: "disabled"
```

Reply:
Added these details. Thank you!